app/routes/_index.tsx

```diff
@@ -1,15 +1,17 @@
-import { json } from "@remix-run/node";
+import { unstable_defineLoader } from "@remix-run/node";
 import type { MetaFunction } from "@remix-run/react";
 import { useLoaderData } from "@remix-run/react";
 
 import { EventDetails } from "~/components/EventDetails";
 import { PageGrid } from "~/components/PageGrid";
 import { constructSiteTitle } from "~/utils/common";
 
-export const loader = async () => {
+import eventsJson from "../data/events.json";
+
+export const loader = unstable_defineLoader(async () => {
```

Comment:
This new `unstable_defineLoader` stuff is strictly for type inference - https://remix.run/docs/en/main/guides/single-fetch#loaderaction-definition-utilities